flink-runtime/src/main/java/org/apache/flink/runtime/security/token/DefaultDelegationTokenManager.java

```
@@ -45,91 +42,86 @@
             @Nullable ScheduledExecutor scheduledExecutor,
             @Nullable ExecutorService ioExecutor) {
         this.configuration = checkNotNull(configuration, "Flink configuration must not be null");
-        this.tokensRenewalTimeRatio = configuration.get(KERBEROS_TOKENS_RENEWAL_TIME_RATIO);
+        this.tokensRenewalTimeRatio = configuration.get(DELEGATION_TOKENS_RENEWAL_TIME_RATIO);
         this.renewalRetryBackoffPeriod =
-                configuration.get(KERBEROS_TOKENS_RENEWAL_RETRY_BACKOFF).toMillis();
-        this.kerberosLoginProvider = new KerberosLoginProvider(configuration);
+                configuration.get(DELEGATION_TOKENS_RENEWAL_RETRY_BACKOFF).toMillis();
         this.delegationTokenProviders = loadProviders();
         this.scheduledExecutor = scheduledExecutor;
         this.ioExecutor = ioExecutor;
     }
 
-    private Map<String, HadoopDelegationTokenProvider> loadProviders() {
+    private Map<String, DelegationTokenProvider> loadProviders() {
         LOG.info("Loading delegation token providers");
 
-        ServiceLoader<HadoopDelegationTokenProvider> serviceLoader =
-                ServiceLoader.load(HadoopDelegationTokenProvider.class);
+        ServiceLoader<DelegationTokenProvider> serviceLoader =
+                ServiceLoader.load(DelegationTokenProvider.class);
 
-        Map<String, HadoopDelegationTokenProvider> providers = new HashMap<>();
-        for (HadoopDelegationTokenProvider provider : serviceLoader) {
+        Map<String, DelegationTokenProvider> providers = new HashMap<>();
+        for (DelegationTokenProvider provider : serviceLoader) {
             try {
                 if (isProviderEnabled(provider.serviceName())) {
                     provider.init(configuration);
                     LOG.info(
                             "Delegation token provider {} loaded and initialized",
                             provider.serviceName());
+                    checkState(
```

What happens if multiple jars contain the `same` provider.
Or a shaded one? Or different versions of the same provider?

---

If 2 providers are coming w/ the same name then it blows up and deployment fails fast.

---

If we wouldn't do that then 2 providers would overwrite each others tokens. Race is never a good thing especially when it's hidden.

---

I am fine with the solution, I was just interested in how this is solved in other places